lib/with_advisory_lock/concern.rb

```diff
@@ -33,7 +33,11 @@ def impl_class
         if adapter.postgresql?
           WithAdvisoryLock::PostgreSQL
         elsif adapter.mysql?
-          WithAdvisoryLock::MySQL
+          if adapter.mysql_nested_lock_support?
```

Comment:
We need a opt-out configuration to not use nested locks even if the adapter support it.